### ISSUE TITLE
Unable to package on Python 3.{10,11,12} with CFFI 1.16 on macOS 14.4 with Clang 15 

### ISSUE THREAD

**Issue by mpasternak** [carsales/pyheif#95]
Hello, 

### **TL;DR:** 
I am unable to build pyheif using ``python setup.py build`` (fails on ``cffi_modules``) on my macOS Sonoma 14.4, but everything is fine if I compile by hand (calling ``python libheif/libheif_build.py`` then copying ``_libheif_cffi*`` from the root dir directly to the virutalenv works fine and gives good results - HEIC files are being open properly). 

I will edit this shortly to add a link to my repo which kindly creates a hack for this. Maybe I will even upload some binary packages to PyPI. 

### Full version:
This looks like setuptools or CFFI setuptools extension bug to me, so this should not be actually reported here. But where? I have no idea, I have no experience of talking to setuptools people, I am posting this in hope "somebody does something" as in contacting some proper maintainer who would show the way, what are we doing wrong. Or, perhaps a version or two passes and somebody fixes this. 

This is the build log:
[pyheif-0.7.1-build-log-fail-python3.12-macos14.4-libffi1.17.txt](https://github.com/carsales/pyheif/files/14719856/pyheif-0.7.1-build-log-fail-python3.12-macos14.4-libffi1.17.txt)

### Issue in progress!
More details will be added later and I will remove this notice. I need to upload my repo first and get a link to this issue...
